packages/use-intl/src/core/createBaseTranslator.tsx

```diff
@@ -276,14 +215,29 @@ function createBaseTranslatorImpl<
     }
 
     try {
-      const formattedMessage = messageFormat.format(
-        // @ts-expect-error `intl-messageformat` expects a different format
-        // for rich text elements since a recent minor update. This
-        // needs to be evaluated in detail, possibly also in regards
-        // to be able to format to parts.
-        prepareTranslationValues({...defaultTranslationValues, ...values})
+      const allValues = {...defaultTranslationValues, ...values};
+      // TODO: The return type seems to be a bit off, not sure if
```

Comment:
there are two different ways to run the precompiled icu messages:

1. `run`
2. `evaluateAst`

the idea is that `run` will always return a string and `evaluateAst` will always return an array.

```tsx
const t = (key, args) => run(messages[key], lang, args);
```

`icu-to-json` is framework independent so it would require you to wrap a react Fragment around the result:

```tsx
t.rich = (key, args) => <>{evaluateAst(messages[key], lang, args)}</>;
```